packages/synapse-react-client/src/synapse-queries/KeyFactory.test.ts

```diff
@@ -0,0 +1,596 @@
+import { KeyFactory } from './KeyFactory'
+import { QueryKey } from '@tanstack/react-query'
+import { MOCK_ACCESS_TOKEN } from '../mocks/MockSynapseContext'
+import { EntityType } from '@sage-bionetworks/synapse-types'
+import {
+  expectQueryKeyToMatch,
+  expectQueryKeyNotToMatch,
+} from './QueryMatchingTestUtils'
+import {
+  BUNDLE_MASK_QUERY_COUNT,
+  BUNDLE_MASK_QUERY_RESULTS,
+} from '../utils/SynapseConstants'
+
+describe('KeyFactory tests', () => {
+  let keyFactory = new KeyFactory(MOCK_ACCESS_TOKEN)
+  beforeEach(() => {
+    keyFactory = new KeyFactory(MOCK_ACCESS_TOKEN)
+  })
+
+  describe('Entity QueryKeys', () => {
```

Comment:
Added many tests to help prevent regressions. Not yet comprehensive

Reply:
Nice "start"!